pkg/util/azblob/manager.go

```diff
@@ -0,0 +1,71 @@
+package azblob
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	azstorage "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/storage/armstorage"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureclient"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/azuresdk/armstorage"
+)
+
+type Manager interface {
+	GetContainerProperties(ctx context.Context, resourceGroupName string, accountName string, containerName string) (azstorage.AccountsClientGetPropertiesResponse, error)
+	GetAZBlobClient(blobContainerURL string, options *azblob.ClientOptions) (AZBlobClient, error)
```

Comment:
I think to avoid issues in FairFax (or other non-public clouds as the ARO service expands), the cloud probably needs to be included in the `ClientOptions` wherever this function is called. I see two places:

1. https://github.com/Azure/ARO-RP/blob/66f608e2ed092e7f26556dffb585bf481f7436fe/pkg/cluster/delete.go#L449
2. https://github.com/Azure/ARO-RP/blob/66f608e2ed092e7f26556dffb585bf481f7436fe/pkg/cluster/deploybaseresources.go#L64

One way I see that you could do this is to pass a reusable `ClientOptions` to the `azblob.Manager` on https://github.com/Azure/ARO-RP/blob/66f608e2ed092e7f26556dffb585bf481f7436fe/pkg/cluster/cluster.go#L195 and then use that `ClientOptions` whenever you instantiate a new client with `GetAZBlobClient`. You can use this one: https://github.com/Azure/ARO-RP/blob/66f608e2ed092e7f26556dffb585bf481f7436fe/pkg/cluster/cluster.go#L173.

Let me know what you think about this.

Reply:
More info about the cloud configuration: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud

Reply:
Wait, false alarm. I see now where the cloud is being pulled from the ARO env: https://github.com/Azure/ARO-RP/blob/66f608e2ed092e7f26556dffb585bf481f7436fe/pkg/util/azureclient/azuresdk/armstorage/accounts.go#L31.

I think we're good here.